### DISCUSSION TITLE
glibc 2.29 is not found on linux 8

### DISCUSSION THREAD

**Discussion by mohsayed99** [abetlen/llama-cpp-python#1846] [Q&A]
On Linux RHEL 8, while using lama_cpp_python to run llama model, I got the following error

OSError: /lib64/libm.so.6: version GLIBC_2.29' not found (required by /d01/miniconda3/envs/textgen/lib/python3.11/site-packages/llama_cpp/lib/libllama.so)

glibc_2.28 is the latest for Linux 8, and cannot be upgraded.
I decided to use lower version of lama_cpp that requires lower glibc, but don't know which version should I use

**Comment by gyanu2507** [comment]
The wheel you installed was built on a newer glibc than RHEL 8's 2.28. `libllama.so` inside the pip package has a `GLIBC_2.29` versioned symbol. You cannot upgrade glibc on RHEL 8, and you cannot "pick an older llama-cpp-python wheel" with much confidence because those wheels are not labeled by glibc.

Build against the system libc:

```bash
CMAKE_ARGS="-DGGML_CUDA=off" pip install llama-cpp-python --no-binary llama-cpp-python --force-reinstall
```

(Add your CUDA/Metal flags as needed.) `--no-binary` compiles `libllama.so` on that box, so it links to 2.28.

If compile is too painful, run the stack in a newer user-space (podman image based on UBI9 / Ubuntu 22.04) and keep RHEL 8 as the host. Mixing a prebuilt `libllama.so` from pip with el8 libc will keep raising this.
